### PR TITLE
Templates

### PR DESCRIPTION
**Pull request by JakeCooper**
Adds the "template" command to the CLI. This fetches all the folders from the repository, allows you to visually scan through them, and auto clones and cleans up for you.

https://www.loom.com/share/3605630a33da4809bf9a28779057aadb

Originally added the v4 GH API, but the graphql stuff doesn't work unauthenticated so I've also added a v3 gateway for now.

### CODE REVIEW

File: cmd/template.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+			clearedOptions = append(clearedOptions, file)
+		}
+		file, err := ui.PromptFiles(prompts[i], clearedOptions)
+		if err != nil {
+			return err
+		}
+		dirs = append(dirs, file.Name)
+	}
+	fmt.Println("Fetching Template...")
+	err := exec.Command("git", "clone", "git@github.com:railwayapp/examples.git", "delete-me").Run()
```
Comment (coffee-cup) [inline cmd/template.go:36]: In the goreleaser homebrew config, we should say that our package depends on git

File: cmd/template.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	if err != nil {
+		return err
+	}
+	fmt.Println("Generating Folder...")
+	err = exec.Command("mv", fmt.Sprintf("delete-me/%s", strings.Join(dirs[:], "/")), ".").Run()
+	if err != nil {
+		return err
+	}
+	fmt.Println("Cleaning Up")
+	err = exec.Command("rm", "-rf", "delete-me").Run()
```
Comment (coffee-cup) [inline cmd/template.go:46]: This won't work on Windows. Maybe use a stdlib call instead (same with above `mv` call)

File: ui/prompt.go [hunk 3]
```diff
@@ -62,3 +62,45 @@ func PromptEnvironments(environments []*entity.Environment) (*entity.Environment
...
+
+	promptType := promptui.Select{
+		Label: "Select Language",
+		Items: []string{"Basic", "Blog", "Todo"},
+	}
+
+	_, _, err = promptType.Run()
+	if err != nil {
+		return "", err
+	}
```
Comment (coffee-cup) [inline ui/prompt.go:85]: suggestion: remove these prompts if they are not used

File: cmd/template.go (hunk 2 above)
Comment (JakeCooper) [reply cmd/template.go:46]: Good calls

**Comment by JakeCooper** [comment]
Closing for now can re-open when we want to explore templating further

**Pull request closed without merging**